From stern@rowland.harvard.edu Wed Aug 10 12:21:18 2005
Date: Wed, 10 Aug 2005 15:15:57 -0400 (EDT)
From: Alan Stern <stern@rowland.harvard.edu>
To: Greg KH <greg@kroah.com>
Cc: <kay.sievers@vrfy.org>
Subject: USB: Fix regression in core/devio.c
Message-ID: <Pine.LNX.4.44L0.0508101510130.4485-100000@iolanthe.rowland.org>

This patch (as551) fixes another little problem recently added to the
USB core.  Someone didn't fix the type of the first argument to
unregister_chrdev_region.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/devio.c |   15 ++++++++-------
 1 files changed, 8 insertions(+), 7 deletions(-)

--- gregkh-2.6.orig/drivers/usb/core/devio.c	2005-08-10 14:41:16.000000000 -0700
+++ gregkh-2.6/drivers/usb/core/devio.c	2005-08-10 14:41:32.000000000 -0700
@@ -76,6 +76,8 @@
 			dev_info( dev , format , ## arg);	\
 	} while (0)
 
+#define USB_DEVICE_DEV		MKDEV(USB_DEVICE_MAJOR, 0)
+
 
 #define	MAX_USBFS_BUFFER_SIZE	16384
 
@@ -1530,18 +1532,17 @@
 {
 	int retval;
 
-	retval = register_chrdev_region(MKDEV(USB_DEVICE_MAJOR, 0),
-				        USB_DEVICE_MAX, "usb_device");
+	retval = register_chrdev_region(USB_DEVICE_DEV, USB_DEVICE_MAX,
+			"usb_device");
 	if (retval) {
 		err("unable to register minors for usb_device");
 		goto out;
 	}
 	cdev_init(&usb_device_cdev, &usbfs_device_file_operations);
-	retval = cdev_add(&usb_device_cdev,
-			  MKDEV(USB_DEVICE_MAJOR, 0), USB_DEVICE_MAX);
+	retval = cdev_add(&usb_device_cdev, USB_DEVICE_DEV, USB_DEVICE_MAX);
 	if (retval) {
 		err("unable to get usb_device major %d", USB_DEVICE_MAJOR);
-		unregister_chrdev_region(USB_DEVICE_MAJOR, USB_DEVICE_MAX);
+		unregister_chrdev_region(USB_DEVICE_DEV, USB_DEVICE_MAX);
 		goto out;
 	}
 	usb_device_class = class_create(THIS_MODULE, "usb_device");
@@ -1550,7 +1551,7 @@
 		retval = PTR_ERR(usb_device_class);
 		usb_device_class = NULL;
 		cdev_del(&usb_device_cdev);
-		unregister_chrdev_region(USB_DEVICE_MAJOR, USB_DEVICE_MAX);
+		unregister_chrdev_region(USB_DEVICE_DEV, USB_DEVICE_MAX);
 	}
 
 out:
@@ -1561,6 +1562,6 @@
 {
 	class_destroy(usb_device_class);
 	cdev_del(&usb_device_cdev);
-	unregister_chrdev_region(USB_DEVICE_MAJOR, USB_DEVICE_MAX);
+	unregister_chrdev_region(USB_DEVICE_DEV, USB_DEVICE_MAX);
 }
 
